lua/neogit/buffers/stash_list_view/init.lua

```diff
@@ -0,0 +1,60 @@
+local Buffer = require("neogit.lib.buffer")
+local config = require("neogit.config")
+local CommitViewBuffer = require("neogit.buffers.commit_view")
+
+local ui = require("neogit.buffers.stash_list_view.ui")
+
+---@class StashListBuffer
+---@field stashes StashEntry[]
+local M = {}
+M.__index = M
+
+--- Gets all current stashes
+function M.new(stashes)
+  local instance = {
+    stashes = stashes,
+  }
+
+  setmetatable(instance, M)
+  return instance
+end
+
+function M:close()
+  self.buffer:close()
+  self.buffer = nil
+end
+
+--- Creates a buffer populated with output of `git stash list`
+--- and supports related operations.
+function M:open()
+  self.buffer = Buffer.create {
+    name = "NeogitStashView",
+    filetype = "NeogitStashView",
+    kind = config.values.stash.kind,
+    context_higlight = true,
+    -- Define the available mappings here. `git stash list` has the same options
+    -- as `git log` refer to git-log(1) for more info.
+    mappings = {
+      n = {
+        ["q"] = function()
+          self:close()
+        end,
+        ["<esc>"] = function()
+          self:close()
+        end,
+        ["<enter>"] = function()
+          -- Still looking for how to view a stash
```

Comment:
You can use https://github.com/NeogitOrg/neogit/blob/f10084ae6ad75b6c7169dabde70d95e59a0b4c33/lua/neogit/lib/git/rev_parse.lua#L23 (`git rev-parse 'stash@{0}'`) to get the OID from a stash, which can be passed into the buffer.